presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java

```diff
@@ -62,6 +73,7 @@
         this.tableName = requireNonNull(tableName, "tableName is null");
         this.compactEffectivePredicate = requireNonNull(compactEffectivePredicate, "compactEffectivePredicate is null");
         this.queue = new AsyncQueue<>(maxOutstandingSplits, executor);
+        this.maxOutstandingSplitsBytes = Ints.checkedCast(maxOutstandingSplitsSize.toBytes());
```

Comment:
Use `toIntExact`

Reply:
Using `Ints.saturatedCast()` might be better here. It means the memory limit will be lower than the very large limit that was specified.